src/components/FilterSidebar/components/QAFilter.test.tsx

```diff
@@ -0,0 +1,73 @@
+import { FilterData } from '@/context/FilterContext/types'; // adjust if needed
+import { render, screen, userEvent } from '@/utils/renderers';
+
+import { QAFilter } from './QAFilter'; // adjust the import based on your file structure
+
+// Mock data for the FilterData type
+const filterData: FilterData = {
+  name: 'Test Filter',
+  type: 'combobox',
+  options: [
+    { value: 'pass', label: 'Pass' },
+    { value: 'partial', label: 'Partial' },
+    { value: 'fail', label: 'Fail' },
```

Comment:
Should also test with the 'none' option as well

Reply:
I can see it as an option [here](https://github.com/EO-DataHub/eodhp-resource-catalogue-ui/pull/17/files#diff-7c2a224ea560b6b43a8d0b9a34fc9aa304c079f669d633055cd2e3333df64295R42)

Reply:
That option should never be used, that is just a value used for `Please select an option`. There are only 2 possibilities:

* Auto-select a real value
* Have a fake option with the label informing the user to select one and check it has been sent when submitting.

I don't want to be adding error handling at every option, especially if it is data-driven as you've built the sidebar i.e. mock `Product Type`, `Licence` etc. If it is dynamic, how do we validate the options selected?

I will change it so that the first option is auto-selected, that way the filter is never invalid, which is probably better than having a fake option, but some prefer that is all.

Reply:
> If it is dynamic, how do we validate the options selected?

Yeah it's a big grey area for me as well, I split it into the input types, so at least we can do some very basic validation based upon the type. 

However, it seems like this validation logic shouldn't lie in the frontend anyway, when whomever gets around to building this microservice that can return filters based upon the data, then perhaps that can also send some type of validation checks.